deadline-monitor.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python
+
+import subprocess
+import json
+import datetime
+import time
+import re
+import traceback
+import sys
+from optparse import OptionParser
+
+class PVC(object):
+    def __init__(self, pvc_name, namespace):
+        self.namespace = namespace
+        self.pvc_name = pvc_name
+
+    def get_pv_name(self):
+        output = subprocess.check_output(['oc', 'get', 'pvc', self.pvc_name, '-n', self.namespace, '-o', 'json'])
+        pvc_json = json.loads(output)
+        return pvc_json['spec']['volumeName']
+
+
+class Pod(object):
+    def __init__(self, pod_data):
+        self.pod_data = pod_data
+        self.pod_name = pod_data['metadata']['name']
+        self.namespace = pod_data['metadata']['namespace']
+        self.uid = pod_data['metadata']['uid']
+
+    def pvc_names(self):
+        pvc_claims = []
+        for volume in self.pod_data['spec']['volumes']:
+            if 'persistentVolumeClaim' in volume:
+                pvc_claims.append(PVC(volume['persistentVolumeClaim']['claimName'], self.namespace))
+        return pvc_claims
+
+
+class PodAlert(object):
+    def run(self):
+        deadlined_pods = self.get_deadlined_pods()
+        if len(deadlined_pods) == 0:
+            print "None found. Everything is OK."
+            return
+        for pod in deadlined_pods:
+            pv_check = self.report_pod_pv(pod)
+
+    def get_deadlined_pods(self):
+        """
+        get pods in DeadlineExceeded state
+        """
+        output = subprocess.check_output(['oc', 'get', 'pods', '--all-namespaces', '-o', 'json'])
+        pod_json = json.loads(output)
+        pod_items = pod_json['items']
+        pods = []
+        for pod in pod_items:
+            pod_status = pod['status']
+            if pod_status['phase'] == 'DeadlineExceeded':
```

Comment:
I think we need to check both `phase` and `reason`.

Here's the json I see for a DeadlineExceeded pod:
```json
            "status": {
                "message": "Pod was active on the node longer than the specified deadline",
                "phase": "Failed",
                "reason": "DeadlineExceeded",
                "startTime": "2017-05-01T15:11:34Z"
            }
```

I would suggest changing this code to:
```python
            if pod_status['phase'] == 'Failed' and 'reason' in pod_status and \
               pod_status['reason'] == 'DeadlineExceeded':
```